### PR TITLE
Fix drive failure

### PR DESCRIPTION
**Pull request by Rohit-PX**
Executed with debug logs in torpedo.

### CODE REVIEW

**Comment by harsh-px** [comment]
@adityadani can you review this?

File: drivers/node/node.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ type Driver interface {
 
 	// YankDrive simulates a failure on the provided drive on the given node.
 	// It returns the UUID of the drive which can be used to recover it back
```
Comment (adityadani) [inline drivers/node/node.go]: nit: add a doc describing the 2nd argument being returned here.

File: tests/drive_failure/drive_failure_test.go [hunk 2]
```diff
@@ -52,13 +52,15 @@ var _ = Describe("{DriveFailure}", func() {
...
 				driveInfoMap := make(map[string]string)
+				busInfoMap := make(map[string]string)
 				Step(fmt.Sprintf("induce a failure on all drives on the node %v", nodeWithDrive), func() {
 					for _, driveToFail := range drives {
-						driveID, err := Inst().N.YankDrive(nodeWithDrive, driveToFail, node.ConnectionOpts{
+						driveID, busID, err := Inst().N.YankDrive(nodeWithDrive, driveToFail, node.ConnectionOpts{
 							Timeout:         1 * time.Minute,
 							TimeBeforeRetry: 5 * time.Second,
 						})
 						driveInfoMap[driveToFail] = driveID
```
Comment (adityadani) [inline tests/drive_failure/drive_failure_test.go]: If driveInfoMap is not required anymore, we can get rid of it.
YankDrive can just return the BusID if that is more reliable for the RecoverDrive command.

File: drivers/node/node.go (hunk 1 above)
Comment (adityadani) [reply drivers/node/node.go]: If the UUID is not being used anymore we can get rid of it and just return the busID.

File: tests/drive_failure/drive_failure_test.go [hunk 3]
```diff
@@ -52,13 +52,15 @@ var _ = Describe("{DriveFailure}", func() {
 				})
 
 				driveInfoMap := make(map[string]string)
+				busInfoMap := make(map[string]string)
 				Step(fmt.Sprintf("induce a failure on all drives on the node %v", nodeWithDrive), func() {
 					for _, driveToFail := range drives {
-						driveID, err := Inst().N.YankDrive(nodeWithDrive, driveToFail, node.ConnectionOpts{
+						driveID, busID, err := Inst().N.YankDrive(nodeWithDrive, driveToFail, node.ConnectionOpts{
 							Timeout:         1 * time.Minute,
 							TimeBeforeRetry: 5 * time.Second,
```
Comment (harsh-px) [inline tests/drive_failure/drive_failure_test.go]: Move timeouts to a const at top of file

File: tests/drive_failure/drive_failure_test.go [hunk 4]
```diff
@@ -73,7 +75,7 @@ var _ = Describe("{DriveFailure}", func() {
 
 				Step(fmt.Sprintf("recover all drives and the storage driver"), func() {
 					for _, driveToFail := range drives {
-						err = Inst().N.RecoverDrive(nodeWithDrive, driveToFail, driveInfoMap[driveToFail], node.ConnectionOpts{
+						err = Inst().N.RecoverDrive(nodeWithDrive, driveToFail, busInfoMap[driveToFail], node.ConnectionOpts{
 							Timeout:         2 * time.Minute,
 							TimeBeforeRetry: 5 * time.Second,
```
Comment (harsh-px) [inline tests/drive_failure/drive_failure_test.go]: move timeouts to a const

Comment (Rohit-PX) [reply tests/drive_failure/drive_failure_test.go]: Fixed.

**Comment by Rohit-PX** [comment]
http://70.0.0.41:8080/view/torpedo/view/all/job/torpedo-dev/51/console

File: drivers/volume/portworx/portworx.go [hunk 5]
```diff
@@ -252,6 +252,21 @@ func (d *portworx) RecoverDriver(n node.Node) error {
 		return err
 	}
 
+	t = func() (interface{}, bool, error) {
+		apiNode, err := d.getClusterManager().Inspect(n.VolDriverNodeID)
```
Comment (harsh-px) [inline drivers/volume/portworx/portworx.go]: there was an issue where we weren't detecting the node as not ok even though it was down. Are you fixing that as part of a different change?

Comment (harsh-px) [reply drivers/volume/portworx/portworx.go]: @Rohit-PX let's add the storage status check too in portworx wait driver up implementation.

**Comment by Rohit-PX** [comment]
http://70.0.0.41:8080/view/torpedo/view/all/job/torpedo-dev/61/console 

Logs for VolumeDriverDownAttachedNode

**Pull request merged**